ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -81,49 +86,75 @@ export class AzureTreeDataProvider implements TreeDataProvider<IAzureNode>, Disp
+                    if (thisTree._azureAccount.status === 'Initializing' || thisTree._azureAccount.status === 'LoggingIn') {
+                        nodes = [new AzureNode(undefined, {
+                            label: localize('loadingNode', 'Loading...'),
+                            commandId: loginCommandId,
+                            contextValue: 'azureCommandNode',
+                            id: loginCommandId,
+                            iconPath: {
+                                light: path.join(__filename, '..', '..', '..', '..', 'resources', 'light', 'Loading.svg'),
+                                dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'dark', 'Loading.svg')
+                            }
+                        })];
+                    } else if (thisTree._azureAccount.status === 'LoggedOut') {
+                        nodes = [
+                            new AzureNode(undefined, { label: localize('signInNode', 'Sign in to Azure...'), commandId: loginCommandId, contextValue: 'azureCommandNode', id: loginCommandId }),
+                            new AzureNode(undefined, { label: localize('createNode', 'Create a Free Azure Account...'), commandId: createCommandId, contextValue: 'azureCommandNode', id: createCommandId })
+                        ];
+                    } else if (thisTree._azureAccount.filters.length === 0) {
+                        commandLabel = localize('noSubscriptionsNode', 'No subscriptions found. Edit filters...');
+                        nodes = [new AzureNode(undefined, { label: commandLabel, commandId: 'azure-account.selectSubscriptions', contextValue: 'azureCommandNode', id: 'azure-account.selectSubscriptions' })];
                     } else {
-                        return new SubscriptionNode(this, this._resourceProvider, subscriptionInfo.subscription.subscriptionId, subscriptionInfo.subscription.displayName, subscriptionInfo);
+                        thisTree._subscriptionNodes = thisTree._azureAccount.filters.map((subscriptionInfo: AzureSubscription) => {
+                            if (subscriptionInfo.subscription.subscriptionId === undefined || subscriptionInfo.subscription.displayName === undefined) {
+                                throw new ArgumentError(subscriptionInfo);
+                            } else {
+                                return new SubscriptionNode(thisTree, thisTree._resourceProvider, subscriptionInfo.subscription.subscriptionId, subscriptionInfo.subscription.displayName, subscriptionInfo);
+                            }
+                        });
+                        nodes = thisTree._subscriptionNodes;
                     }
-                });
-                nodes = this._subscriptionNodes;
-            }
 
-            return nodes.concat(this._rootNodes);
+                    result = nodes.concat(thisTree._rootNodes);
+                }
+
+                this.measurements.childCount = result.length;
```

Comment:
Do you care about whether it has more children, or whether this request is a result of load more children?

Reply:
I'll add something about 'hasMoreChildren'. Not really interested in load more since we already have a command tracking loadMore